schema.graphql

```diff
@@ -2329,3 +2358,157 @@ type CustomClaimType @entity {
   createdBlock: Block!
   updatedBlock: Block!
 }
+
+enum AffirmingPartyEnum {
+  Sender
+  Receiver
+  Mediator
+}
+
+enum AffirmStatusEnum {
+  Affirmed
+  Rejected
+}
+
+enum TransactionStatusEnum {
+  Created
+  Executed
+  Rejected
+}
+
+"""
+Represents the proof of a transaction affirmation
+"""
+type Proof @jsonField {
+  assetId: String!
+  proof: String!
+}
+
+"""
+Represents a confidential account
+"""
+type ConfidentialAccount @entity {
+  id: ID!
+  account: String!
+  creator: Identity!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential venue
+"""
+type ConfidentialVenue @entity {
+  id: ID!
+  venueId: Int!
+  creator: Identity!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Used for many to many relation between ConfidentialAsset and ConfidentialVenue
+"""
+type ConfidentialAssetAllowedVenue @entity {
+  id: ID!
+  asset: ConfidentialAsset!
+  venue: ConfidentialVenue!
+  isAllowed: Boolean!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
```

Comment:
I don't think we need a separate entity for this. We can have `allowedVenues: [Int!]` within ConfidentialAsset itself.